src/test/test-sleep.c

```diff
@@ -18,13 +18,41 @@
   along with systemd; If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#include <linux/fiemap.h>
 #include <stdio.h>
 
+#include "fd-util.h"
 #include "log.h"
 #include "sleep-config.h"
 #include "strv.h"
 #include "util.h"
 
+static int test_fiemap(void) {
+        _cleanup_free_ struct fiemap *fiemap = NULL;
+        _cleanup_close_ int fd = -1;
+        int r;
+        const char *device = ABS_BUILD_DIR "/systemd-sleep";
+
+        fd = open(device, O_RDONLY | O_CLOEXEC | O_NONBLOCK);
+        if (fd < 0)
+                return log_error_errno(errno, "failed to open %s: %m", device);
+        log_info("fd: %d", fd);
+        r = read_fiemap(fd, &fiemap);
+        if (r < 0)
+                return log_error_errno(r, "Unable to read extent map for '%s': %m",
+                                       device);
+        log_info("extent map information for %s:", device);
+        log_info("\t start: %llu", fiemap->fm_start);
+        log_info("\t length: %llu", fiemap->fm_length);
+        log_info("\t flags: %u", fiemap->fm_flags);
+        log_info("\t number of mapped extents: %u", fiemap->fm_mapped_extents);
+        log_info("\t extent count: %u", fiemap->fm_extent_count);
+        log_info("\t first extent location: %llu",
+                 fiemap->fm_extents[0].fe_physical / page_size());
```

Comment:
Thanks for this.

Hmm, here I see:
```
extent map information for /home/zbyszek/src/systemd/build/systemd-sleep:
	 start: 94545859689904
	 length: 8299960843193248865
	 flags: 1702130553
	 number of mapped extents: 1
	 extent count: 1885693292
	 first extent location: 12099829
```
Are those large numbers exptected?
(This is on ext4.)